toolbox/fdc3-conformance/Intents-Tests.md

```diff
@@ -0,0 +1,283 @@
+
+### Raise Intent (Ignoring any result)
+
+| App   | Step           | Details                                                                                           |
+|-------|----------------|---------------------------------------------------------------------------------------------------|
+| Test  | 1. Raise        | `fdc3.raiseIntent("aTestingIntent", testContextX)`<br>starts app A.                       |
+| A     | 2. Receive Intent & Context | After starting up, A runs `fdc3.addIntentListener("aTestingIntent1")` to register its listener.<br>It then receives `testContextX`, matching that sent by Test |
+| Test  | 3. IntentResolution          | The `raiseIntent` call returns an `IntentResolution` Object with an `AppIdentifer` as the `source field` with App A's `appId` and `instanceId` set.**                       |
+
+- `2.0-RaiseIntentSingleResolve`: Perform above test
+- `2.0-RaiseIntentTargetedAppResolve`: Repeat the above test, but:
+  - In the first step use `fdc3.raiseIntent("sharedTestingIntent1", testContextX, {"appID": "<B’s appId>"})` to start app B,
+  - Otherwise, as above.
+- `2.0-RaiseIntentTargetedInstanceResolveOpen`: Repeat the above test, but:
+  - Before the first step, use `let appIdentifier = await fdc3.open({appId: "<A's appId>"})` to start A and retrieve its `AppIdentifier` with instance details.
+  - Then in the first step, use `fdc3.raiseIntent("aTestingIntent", testContextX, appIdentifier)` to target the running instance of app A. 
+  - Confirm that the intent is delivered to the correct instance and that another instance is NOT started. Otherwise, as above.
+- `2.0-RaiseIntentTargetedInstanceResolveFindInstances`: Repeat the above test, but:
+  - Before the first step, use `let appIdentifier = await fdc3.open({appId: "<A's appId>"})` to start A.
+  - Then use `const instances = await fdc3.findInstances({appId: "<A's appId>"})` to retrieve a list of instances of app A. Confirm that only one is present and retrieve its `AppIdentifier`, confirming that it contains an `instanceId` field that matches that returned by the `fdc3.open` call.
+  - Then in the first step, use `fdc3.raiseIntent("aTestingIntent", testContextX, appIdentifier)` to target the running instance of app A. 
+  - Confirm that the intent is delivered to the correct instance and that another instance is NOT started. Otherwise, as above.
+- `2.0-RaiseIntentFailedResolve`: Perform above test, but:
+  - Use `fdc3.raiseIntent("aTestingIntent", testContextY)`.  Note that no app supports this intent and context combination.**
+  - You should receive a JavaScript Error with the message `ResolveError.NoAppsFound`.
+- `2.0-RaiseIntentFailTargetedAppResolve1`: Perform above test, but:
+  - Use `fdc3.raiseIntent("aTestingIntent", testContextY, {appId: "<A's appId>"})`.
+  - You should receive a JavaScript Error with the message `ResolveError.IntentDeliveryFailed`.
```

Comment:
Expected error must be `ResolveError.NoAppsFound` (since A doesn't support this context type) - same as RaiseIntentFailTargetedAppResolve1 test

Reply:
@ksgeorgieva is correct. The semantics of that error are not great, due to it coming from an earlier era, but it should be returned whenever the resolver can't find a match between intent/context and an app. While `IntentDeliveryFailed` is used when it can find one, but fails to deliver the intent/context payload to it. `TargetAppUnavailable` is used if an appId is given but not recognised and `TargetInstanceUnavailable` if an instanceId is not recognised.

```suggestion
  - You should receive a JavaScript Error with the message `ResolveError.NoAppsFound`.
```